storage/src/main/java/org/apache/kafka/storage/internals/log/LogAppendInfo.java

```diff
@@ -117,7 +117,7 @@ public LogAppendInfo(long firstOffset,
         this.lastOffset = lastOffset;
         this.lastLeaderEpoch = lastLeaderEpoch;
         this.maxTimestamp = maxTimestamp;
-        this.offsetOfMaxTimestamp = offsetOfMaxTimestamp;
+        this.shallowOffsetOfMaxTimestamp = offsetOfMaxTimestamp;
```

Comment:
Could we rename offsetOfMaxTimestamp in the input param and the javadoc accordingly? It would be useful to rename the local val in `UnifiedLog.analyzeAndValidateRecords` too.